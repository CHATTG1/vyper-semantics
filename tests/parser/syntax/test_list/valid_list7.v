%pgm(,
  %svdecl(y, %listT(%decimal, 3), %private),
  %fdecl(%@public, foo, %param(x, %listT(%num, 3)), %void,
    %assign(%svar(y), %var(x)))
)
