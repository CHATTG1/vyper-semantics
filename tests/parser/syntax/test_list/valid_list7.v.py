y: decimal[3]

@public
def foo(x: num[3]):
    self.y = x
