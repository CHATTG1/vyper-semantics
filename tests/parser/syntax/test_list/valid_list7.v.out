$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3109404179 ) , $%seq ( $calldatacopy ( 320 , 4 , 96 ) , $assert ( $iszero ( $callvalue ) ) , $%seq ( $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 32 ) ) , $mload ( 64 ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 64 ) ) , $mload ( 64 ) ) ) , $with ( $_L , $sha3_32 ( 0 ) , $with ( $_R , 320 , $%seq ( $sstore ( $add ( $_L , $uclamplt ( 0 , 3 ) ) , $mul ( $mload ( $add ( $mul ( 32 , $uclamplt ( 0 , 3 ) ) , $_R ) ) , 10000000000 ) ) , $sstore ( $add ( $_L , $uclamplt ( 1 , 3 ) ) , $mul ( $mload ( $add ( $mul ( 32 , $uclamplt ( 1 , 3 ) ) , $_R ) ) , 10000000000 ) ) , $sstore ( $add ( $_L , $uclamplt ( 2 , 3 ) ) , $mul ( $mload ( $add ( $mul ( 32 , $uclamplt ( 2 , 3 ) ) , $_R ) ) , 10000000000 ) ) ) ) ) , $stop ) ) ) , 0 ) ) )
